src/gen_lb.erl

```diff
@@ -258,30 +255,43 @@ cluster_maybe_block(Up0, Down, MaxErrors, MaxCrasches) ->
                               ({call_crash, _}, {E,C}) -> {E,  C+1}
                            end, {0, 0}, Info) of
             {E, _C} when MaxErrors =/= 0, E >= MaxErrors ->
-              ?warning("blocking ~p (too many errors)", [Node]),
+              ?LOG_WARNING("blocking ~p (too many errors)", [Node]),
               false;
             {_E, C} when MaxCrasches =/= 0, C >= MaxCrasches ->
-              ?warning("blocking ~p (too many crasches)", [Node]),
+              ?LOG_WARNING("blocking ~p (too many crasches)", [Node]),
               false;
             {_, _} ->
               true
           end
       end, Up0),
-  Now = s2_time:stamp() div 1000,
+  Now = os:system_time(millisecond),
   {Up, Down ++ [{Node, Now} || {Node, _Info} <- NewDown]}.
 
 cluster_maybe_unblock(Up, Down, BlockTime) ->
-  Timeout = (s2_time:stamp() div 1000) - BlockTime,
+  Timeout = os:system_time(millisecond) - BlockTime,
   {NewUp, NewDown} =
     lists:partition(
       fun({_Node,  inf })     -> false;
          ({ Node,  Time})
-          when Time < Timeout -> ?warning("unblocking ~p", [Node]),
+          when Time < Timeout -> ?LOG_WARNING("unblocking ~p", [Node]),
                                  true;
          ({_Node, _Time})     -> false
       end, Down),
   {Up ++ [{Node, []} || {Node, _Time} <- NewUp], NewDown}.
 
+
+assoc(L, K) ->
+  case lists:keyfind(K, 1, L) of
+    {K, V} -> {ok, V};
+    false  -> {error, notfound}
+  end.
+
+assoc(KVs, K, Def) ->
+  case assoc(KVs, K) of
+    {ok, V}           -> V;
+    {error, notfound} -> Def
+  end.
```

Comment:
I prefer self describing variable names, aka not shorten. The one I found most confusing of these is `Def`, first I read `Definition` and the. `Default`. Please expand these, we can afford the extra bytes 😉